stdlib/2and3/shutil.pyi

```diff
@@ -80,39 +102,25 @@ if sys.version_info >= (3, 3):
     def which(cmd: _Path, mode: int = ...,
               path: Optional[_Path] = ...) -> Optional[str]: ...
 
-if sys.version_info >= (3, 4):
-    class Error(OSError): ...
-    class SameFileError(Error): ...
-    class SpecialFileError(OSError): ...
-    class ExecError(OSError): ...
-    class ReadError(OSError): ...
-else:
-    class Error(EnvironmentError): ...
-    class SpecialFileError(EnvironmentError): ...
-    class ExecError(EnvironmentError): ...
-    class ReadError(EnvironmentError): ...
-class RegistryError(Exception): ...
-
-def make_archive(base_name: str, format: str, root_dir: _Path = ...,
+def make_archive(base_name: _AnyStr, format: str, root_dir: _Path = ...,
                  base_dir: _Path = ..., verbose: bool = ...,
                  dry_run: bool = ..., owner: str = ..., group: str = ...,
-                 logger: Any = ...) -> str: ...
+                 logger: Any = ...) -> _AnyStr: ...
 def get_archive_formats() -> List[Tuple[str, str]]: ...
 
-# TODO function is a callback that receives keyword arguments; should make it not use Any
-# once we have support for callable types with keyword args
-def register_archive_format(name: str, function: Any,
+def register_archive_format(name: str, function: Callable[..., Any],
                             extra_args: Sequence[Tuple[str, Any]] = ...,
```

Comment:
extra_args is Optional.

Also, the implementation actually allows only tuples and lists, not all sequences (there's an isinstance check). And within the sequence, the elements can be either tuples or lists. The first change would be a pain to support properly due to invariance, but we can accommodate the second one.

In summary, I think this should be `Optional[Sequence[Union[Tuple[str, Any], List[Any]]]`.